### PR TITLE
Own chat renamed to saved messages (#1548)

### PR DESCRIPTION
**Pull request by azzimuth**
Solves #1548

### CODE REVIEW

File: app/js/controllers.js [hunk 1]
```diff
@@ -1316,14 +1320,15 @@ angular.module('myApp.controllers', ['myApp.i18n'])
       }
 
       peerHistory = historiesQueuePush(peerID)
-
+
```
Comment (stek29) [inline app/js/controllers.js]: trailing whitespace :0

File: app/js/controllers.js [hunk 2]
```diff
@@ -1316,14 +1320,15 @@ angular.module('myApp.controllers', ['myApp.i18n'])
...
-        data: peerData
-      })
-
+        data: peerData,
+        self: peerID == AppUsersManager.getSelf().id
+      })      
+         
       MtpApiManager.getUserID().then(function (myID) {
-        $scope.ownID = myID
+        $scope.ownID = myID
```
Comment (stek29) [inline app/js/controllers.js]: same, trailing whitespace

File: app/js/directives.js [hunk 3]
```diff
@@ -3100,14 +3100,16 @@ angular.module('myApp.directives', ['myApp.filters'])
       var override = attrs.userOverride && $scope.$eval(attrs.userOverride) || {}
       var short = attrs.short && $scope.$eval(attrs.short)
       var username = attrs.username && $scope.$eval(attrs.username)
-
-      var peerID
+      var dialogList = attrs.dialogList && $scope.$eval(attrs.dialogList) || false
+      var myID = AppUsersManager.getSelf().id
+
```
Comment (stek29) [inline app/js/directives.js]: same :)

File: app/js/services.js [hunk 4]
```diff
@@ -189,6 +189,14 @@ angular.module('myApp.services', ['myApp.i18n', 'izhukov.utils'])
       return getUser(myID)
     }
 
+    function getSavedMessages () {      
+      var apiUser = jQuery.extend(true, {}, getUser(myID))
```
Comment (stek29) [inline app/js/services.js]: I guess angular.copy would be better

File: app/less/desktop.less [hunk 5]
```diff
@@ -217,18 +217,30 @@
...
+  &-saved-messages {
+    width: 16px;
+    height: 18px;
+    margin-right: 42px;
+    display: inline-block;
+    vertical-align: top;
+    margin-top: 3px;
+    .image-2x('../img/icons/General.png', 40px, 972px);
+    background-position: -10px -938px;
+
```
Comment (stek29) [inline app/less/desktop.less]: An empty line :)

File: app/js/services.js [hunk 6]
```diff
@@ -189,6 +189,14 @@ angular.module('myApp.services', ['myApp.i18n', 'izhukov.utils'])
...
+        apiUser.rFirstName = _('user_first_name_self')
```
Comment (stek29) [inline app/js/services.js]: Indentation is weird :)

File: app/js/services.js [hunk 7]
```diff
@@ -189,6 +189,14 @@ angular.module('myApp.services', ['myApp.i18n', 'izhukov.utils'])
...
+        apiUser.rFullName = _('user_name_self')
+        apiUser.photo._ = 'self'
```
Comment (stek29) [inline app/js/services.js]: .photo might be missing

Also, when user lacks photo, saved messages icon gets replaced by initials sometimes

File: app/partials/desktop/im.html [hunk 8]
```diff
@@ -206,7 +206,7 @@ <h3 class="im_dialogs_empty_header" my-i18n="im_get_started"></h3>
 
                 <div class="im_send_form_wrap clearfix" ng-controller="AppImSendController" ng-class="{im_send_reply_form_wrap: draftMessage.replyToMsgID > 0}">
 
-                  <a class="pull-right im_panel_peer_photo" my-peer-photolink="historyPeer.id" img-class="im_panel_peer_photo" watch="true" ng-show="!draftMessage.isBroadcast">
+                  <a class="pull-right im_panel_peer_photo" my-peer-photolink="historyPeer.id" img-class="im_panel_peer_photo" watch="true" ng-show="!draftMessage.isBroadcast" ng-if="!historyPeer.self">
```
Comment (stek29) [inline app/partials/desktop/im.html]: Instead of just hiding it completely would be better to show "Saved messages" (and open profile on click?)

**Comment by stek29** [comment]
Great PR 🎉 

What about going further and making forwarded messages look as they do on other apps, and adding "go to original message"? (for this one scheme has to be updated, but won't be that hard), and always showing saved messages on top when forwarding?

**Comment by stek29** [comment]
Oh, suppressed by 90380025f6bb4c6eb149c6343c64a87680adca8f 
Sorry, and thanks anyway

**Comment by azzimuth** [comment]
@stek29, @zhukov has already implemented this.

@zhukov could you even add new option to menu for Saved Messages (both desktop and mobile) as I did?

**Pull request closed without merging**